src/components/ReportActionItemIOUPreview.js

```diff
@@ -82,6 +83,14 @@ const ReportActionItemIOUPreview = ({
     onPayButtonPressed,
     translate,
 }) => {
+    // Usually the parent determines whether the IOU Preview is displayed. But as the iouReport total cannot be known
+    // until it is stored locally, we need to make this check within the Component after retrieving it. This allows us
+    // to handle the loading UI from within this Component instead of needing to declare it within each parent, which
+    // would duplicate and complicate the logic
```

Comment:
Tbh, I don't really follow what logic would be complicated by this. It seems like the logic is already kind of weird and confusing and this comment doesn't explain much about why it was set up that way.

e.g. this component _perhaps_ should not be responsible for fetching the iou at all...

this comment made me look into how we are doing it and... I think it's a pretty bad practice to be making an API call like this when the component renders... (instead of e.g. using a lifecycle method)

https://github.com/Expensify/Expensify.cash/blob/ec0070aafa0374c2d02bdf9db757e98f10b59aaf/src/components/ReportActionItemIOUPreview.js#L102-L104

Not sure if anything needs to be done right now and I won't block on it (since the issue seems important to address), but feels like a code smell to me.

Reply:
Yeah, when making these changes it became apparent that the parent should perhaps be handling the loading state. Also, only one use of the comp has any requirement for displaying the preview now, so it might be possible to further simplify the `shouldDisplay` logic. 

I didn't want this to turn into a large refactor and risk longer delays, so I've [created an issue](https://github.com/Expensify/Expensify/issues/166564) to refactor this component that I'll make external.